src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralActions.ts

```diff
@@ -94,8 +95,18 @@ export const setDataFromUrl: ActionCreator<ThunkAction<
       })
     : null;
 
-  if (entityId && entityId !== activeEntityId) {
-    dispatch(updateEnsObject(entityId));
+  if (entityId && genomeIdFromUrl) {
+    if (entityId !== activeEntityId) {
+      dispatch(updateEnsObject(entityId));
```

Comment:
I completely forgot that we still are using the notion of "ensObjects" in entity viewer 😞 

Wouldn't you agree that we should change all references to ensObjects in entity viewer to "entities" (unless they are too tightly entangled with ensObjects from genome browser)?

I also noticed that the `updateEnsObject` action fetches the ens object if it's missing from the state. I suspect we don't need this for the Entity Viewer state anymore.

Does this sound like something for a separate PR to you?

Reply:
>Wouldn't you agree that we should change all references to ensObjects in entity viewer to "entities" (unless they are too tightly entangled with ensObjects from genome browser)?
I also noticed that the updateEnsObject action fetches the ens object if it's missing from the state. I suspect we don't need this for the Entity Viewer state anymore.

Yes, that makes sense. `ensObjects` is not used elsewhere in entity viewer except this action so I guess it's not tightly entangled.